clean_build.sh

```diff
@@ -4,26 +4,23 @@ cd "$(dirname "$0")" || exit 1
 
 # some environments have cmake v2 as 'cmake' and v3 as 'cmake3'
 # check for cmake3 first then fallback to just cmake
-
-B_CMAKE=$(command -v cmake3 2>/dev/null)
-if [ "$?" -eq 0 ]; then
-    continue
-else
-    if command -v cmake 2>/dev/null; then
-        B_CMAKE=$(command -v cmake)
-    else
-        echo "ERROR: CMake not in $PATH, cannot build! Please install CMake, or if this persists, file a bug report."
-        exit 1
-    fi
+[ -n "$B_CMAKE" ] || B_CMAKE=$(command -v cmake3)
+[ -n "$B_CMAKE" ] || B_CMAKE=$(command -v cmake2)
+[ -n "$B_CMAKE" ] || B_CMAKE=$(command -v cmake)
+if [ -z "$B_CMAKE" ]; then
+    echo "ERROR: CMake not in $PATH, cannot build! Please install CMake, or if this persists, file a bug report."
+    exit 1
 fi
 
-CMAKE_VERSION="$("$B_CMAKE" --version)"
-if [ "$(printf '%s\n' "3.12.0" "$currentver" | sort -V | head -n1)" = "3.12.0" ]; then
-    CMAKE_HAS_PARALLEL=1
-else
-    CMAKE_HAS_PARALLEL=0
+# Let's be reasonably nice in shared envs like CI runners
+nproc=$(nproc)
```

Comment:
This script is also used on OSX, `nproc` is not available there.

Reply:
fixed by running `sysctl` on darwin